### ISSUE TITLE
Split up `core` into multiple files.

### ISSUE THREAD

**Issue by srush** [danoneata/chalk#89]
The file core.py is getting really big and seems to do too many things. Would like to split up core.py to look more like the diagrams hierarchy (align.py, render.py, juxtopose.py, arrow.py etc.) However the OO style is convenient for the user. 

There isn't a nice way to do this in Python, but this seems not so bad.

https://stackoverflow.com/questions/47561840/how-can-i-separate-the-functions-of-a-class-into-multiple-files

One side benefit of this is that every function could in theiry be used in either functional or OO calling convention. (Like atop now).

**Comment by danoneata** [comment]
Hello! That sounds good! I'm in favour of splitting the `core.py` module in submodules based on their functionality and using a more functional style.

A question that I have is whether we are going to end up with some circular imports and how to fix this. For example, let's say we have a `core/render/svg.py` submodule that defines the `to_svg` and `render_svg` functions. I would expect this module to need importing the data types (for type hinting and maybe also for dispatching based on the data type):
```python
from chalk.types import Diagram, Primitive, Empty

def to_svg(diagram: Diagram, dwg: Drawing, style: Style) -> BaseElement:
    if isinstance(diagram, Primitive):
        ...
    elif isinstance(diagram, Empty):
        ...
    ...

def render_svg(diagram: Diagram, **kwargs):
    ...
    # uses to_svg
```
Conversely, in `core/types.py` I would expect the `render_svg` function to be imported in order to attach it to the `Diagram` object. These mutual imports will generate an `ImportError`; but maybe there is an easy fix to this.

The functions that traverse the `Diagram` abstract syntax tree (like `to_svg` or `to_tikz`), will need to pattern match on the data type when written in a more functional style. There are probably multiple ways of achieving this: `isinstance` checks (as above), [structural pattern matching](https://peps.python.org/pep-0635/) (available only from Python 3.10); [the visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern#Python_example). I'm curious which approach you would prefer.

**Comment by srush** [comment]
Yes, this is a very valid concern. I do not know what the best answer is, and need to do some more research. 

I was hoping to circumvent this problem for now by not factoring rendering. By starting with only functions of the base class of `Diagram`, I think we could get away with just calling other functions on diagram that we import.
